### PR TITLE
UTF-8 strings for universal compilation

### PR DESCRIPTION
**Pull request by jojorne**
Fix error: Unknown character.

If you have your OS in another language, you will not be able to compile.

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/4364)

### CODE REVIEW

**Comment by JosJuice** [comment]
Should the same change be made to the interface config pane? https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp#L57

There aren't any build problems with that one as far as I know, but it would be nice to have readable language names instead of strings of character literals.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by jojorne** [comment]
I didn't know about it, good idea!

File: Source/Core/DolphinQt2/AboutDialog.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2016 Dolphin Emulator Project
+﻿// Copyright 2016 Dolphin Emulator Project
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/AboutDialog.cpp]: It looks like you added a UTF-8 BOM. Could you revert it?

File: Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2015 Dolphin Emulator Project
+﻿// Copyright 2015 Dolphin Emulator Project
```
Comment (leoetlino) [inline Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp]: and probably here too

File: Source/Core/DolphinWX/NetPlay/NetWindow.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2015 Dolphin Emulator Project
+﻿// Copyright 2015 Dolphin Emulator Project
```
Comment (leoetlino) [inline Source/Core/DolphinWX/NetPlay/NetWindow.cpp]: and here

File: Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp [hunk 4]
```diff
@@ -54,35 +54,34 @@ void InterfaceConfigPane::InitializeGUI()
...
+  m_interface_lang_strings.Add(u8"Svenska");			// Swedish
+  m_interface_lang_strings.Add(u8"Türkçe");				// Turkish
+  m_interface_lang_strings.Add(u8"Ελληνικά");			// Greek
+  m_interface_lang_strings.Add(u8"Русский");			// Russian
+  m_interface_lang_strings.Add(u8"العربية");			// Arabic
+  m_interface_lang_strings.Add(u8"فارسی");				// Farsi
+  m_interface_lang_strings.Add(u8"한국어");				// Korean
+  m_interface_lang_strings.Add(u8"日本語");				// Japanese
+  m_interface_lang_strings.Add(u8"简体中文");			// Simplified Chinese
+  m_interface_lang_strings.Add(u8"繁體中文");			// Traditional Chinese
```
Comment (JosJuice) [inline Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp]: Please keep the comments aligned using spaces. We don't use tabs in the Dolphin source code.

**Comment by jojorne** [comment]
Unfortunately without the BOM, Visual Studio does not understand UTF-8 character if the OS is in another language (https://connect.microsoft.com/VisualStudio/Feedback/Details/888437). Is this a problem?

**Comment by JosJuice** [comment]
I don't know if a BOM would cause problems elsewhere, but they're very non-standard in UTF-8, so we shouldn't use them. If VS can't handle UTF-8 without BOM, it might be preferable to just do something like what InterfaceConfigPane did.

**Pull request closed without merging**